src/Data/Array.purs

```diff
@@ -708,6 +1053,12 @@ foldRecM f a array = tailRecM2 go a 0
         pure (Loop { a: res', b: i + 1 })
 
 -- | Find the element of an array at the specified index.
+-- |
+-- | ```purescript
+-- | unsafePartial $ unsafeIndex ["a", "b", "c"] 1 = "b"
+-- | unsafePartial $ unsafeIndex ["a", "b", "c"] 10 -- runtime error
```

Comment:
I think this comment is still not quite correct.

Reply:
I checked again in `psci`.
Result of evaluating:
```purescript
> unsafePartial $ unsafeIndex ["a", "b", "c"] 1
"b"

> unsafePartial $ unsafeIndex ["a", "b", "c"] 10
/home/.../.psci_modules/node_modules/Data.Show/foreign.js:30
  var l = s.length;
            ^

TypeError: Cannot read property 'length' of undefined
    at exports.showStringImpl (/home/.../.psci_modules/node_modules/Data.Show/foreign.js:30:13)
    at /home/.../node_modules/Control.Monad.Eff.Console/index.js:19:53
    at Object.<anonymous> (/home.../.psci_modules/node_modules/$PSCI/index.js:19:84)
    at Module._compile (module.js:641:30)
    at Object.Module._extensions..js (module.js:652:10)
    at Module.load (module.js:560:32)
    at tryModuleLoad (module.js:503:12)
    at Function.Module._load (module.js:495:3)
    at Module.require (module.js:585:17)
    at require (internal/module.js:11:18)
```

Reply:
this is with version:
```bash
$ pulp --version                                                               
Pulp version 12.0.1
purs version 0.11.7 using /usr/local/bin/purs
```

Reply:
Right, but the error message indicates that the error comes from attempting to `Show` it - we might not necessarily do that in other situations.

Reply:
I'll try to come up with a good example

Reply:
Oh yes, I see. Did not think of that..

Reply:
For example:
```purescript
module Main where

import Prelude
import Data.Array as Array
import Partial.Unsafe (unsafePartial)
import Control.Monad.Eff (Eff)
import Control.Monad.Eff.Console (CONSOLE, log)

main :: forall e. Eff (console :: CONSOLE | e) Unit
main = do
  let oops = unsafePartial (Array.unsafeIndex [true] 1)
  log $ if oops
    then "it was true"
    else "it was false"
```

Runs successfully and prints "it was false", because `undefined` is falsy.

Reply:
Since this is a little subtle, maybe it would be best to remove the out-of-range example, and instead explain what happens if you try to `unsafeIndex` with an out-of-range index in prose. How about:

> Using `unsafeIndex` with an out-of-range index will not immediately raise a runtime error. Instead, the result will be `undefined`. Most attempts to subsequently use the result will cause a runtime error, of course, but this is not guaranteed, and is dependent on the backend; some programs will continue to run as if nothing is wrong. For example, in the JavaScript backend, the expression `unsafePartial (unsafeIndex [true] 1)` has type `Boolean`; since this expression evaluates to `undefined`, attempting to use it in an `if` statement will cause the `else` branch to be taken.